R2 bucket creation is broken due to API change

### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the library and it is still present.

### cloudflare-go version

### Expected output

Able to create R2 bucket

### Actual output

✘ [ERROR] A request to the Cloudflare API (/accounts/<account id>/r2/buckets/bucketname) failed.

  No route matches this url. [code: 10015]

### Steps to reproduce

Attempt to create R2 bucket with API

### References

https://github.com/cloudflare/wrangler2/issues/1654

--- a/.changelog/1035.txt
+++ b/.changelog/1035.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+r2: fix create bucket endpoint
+```

--- a/r2_bucket.go
+++ b/r2_bucket.go
@@ -11,20 +11,24 @@ var (
 	ErrMissingBucketName = errors.New("require bucket name missing")
 )
 
+type CreateR2BucketParameters struct {
+	Name string `json:"name,omitempty"`
+}
+
 // CreateR2Bucket Creates a new R2 bucket.
 //
 // API reference: https://api.cloudflare.com/#r2-bucket-create-bucket
-func (api *API) CreateR2Bucket(ctx context.Context, rc *ResourceContainer, bucketName string) error {
+func (api *API) CreateR2Bucket(ctx context.Context, rc *ResourceContainer, params CreateR2BucketParameters) error {
 	if rc.Identifier == "" {
 		return ErrMissingAccountID
 	}
 
-	if bucketName == "" {
+	if params.Name == "" {
 		return ErrMissingBucketName
 	}
 
-	uri := fmt.Sprintf("/accounts/%s/r2/buckets/%s", rc.Identifier, bucketName)
-	_, err := api.makeRequestContext(ctx, http.MethodPost, uri, nil)
+	uri := fmt.Sprintf("/accounts/%s/r2/buckets", rc.Identifier)
+	_, err := api.makeRequestContext(ctx, http.MethodPost, uri, params)
 
 	return err
 }

--- a/r2_bucket_test.go
+++ b/r2_bucket_test.go
@@ -15,7 +15,7 @@ func TestR2_CreateBucket(t *testing.T) {
 	setup()
 	defer teardown()
 
-	mux.HandleFunc(fmt.Sprintf("/accounts/%s/r2/buckets/%s", testAccountID, testBucketName), func(w http.ResponseWriter, r *http.Request) {
+	mux.HandleFunc(fmt.Sprintf("/accounts/%s/r2/buckets", testAccountID), func(w http.ResponseWriter, r *http.Request) {
 		assert.Equal(t, http.MethodPost, r.Method, "Expected method 'POST', got %s", r.Method)
 		w.Header().Set("content-type", "application/json")
 		fmt.Fprintf(w, `{
@@ -26,17 +26,17 @@ func TestR2_CreateBucket(t *testing.T) {
 }`)
 	})
 
-	err := client.CreateR2Bucket(context.Background(), AccountIdentifier(""), "")
+	err := client.CreateR2Bucket(context.Background(), AccountIdentifier(""), CreateR2BucketParameters{})
 	if assert.Error(t, err) {
 		assert.Equal(t, ErrMissingAccountID, err)
 	}
 
-	err = client.CreateR2Bucket(context.Background(), AccountIdentifier(testAccountID), "")
+	err = client.CreateR2Bucket(context.Background(), AccountIdentifier(testAccountID), CreateR2BucketParameters{})
 	if assert.Error(t, err) {
 		assert.Equal(t, ErrMissingBucketName, err)
 	}
 
-	err = client.CreateR2Bucket(context.Background(), AccountIdentifier(testAccountID), "example-bucket")
+	err = client.CreateR2Bucket(context.Background(), AccountIdentifier(testAccountID), CreateR2BucketParameters{testBucketName})
 	assert.NoError(t, err)
 }
 

--- a/r2_bucket_test.go
+++ b/r2_bucket_test.go
@@ -31,12 +31,12 @@ func TestR2_CreateBucket(t *testing.T) {
 		assert.Equal(t, ErrMissingAccountID, err)
 	}
 
-	err = client.CreateR2Bucket(context.Background(), AccountIdentifier(testAccountID), CreateR2BucketParameters{})
+	err = client.CreateR2Bucket(context.Background(), AccountIdentifier(testAccountID), CreateR2BucketParameters{Name: ""})
 	if assert.Error(t, err) {
 		assert.Equal(t, ErrMissingBucketName, err)
 	}
 
-	err = client.CreateR2Bucket(context.Background(), AccountIdentifier(testAccountID), CreateR2BucketParameters{testBucketName})
+	err = client.CreateR2Bucket(context.Background(), AccountIdentifier(testAccountID), CreateR2BucketParameters{Name: testBucketName})
 	assert.NoError(t, err)
 }
 

